File: src/main/java/moe/sota/decompiler/types/ManifestType.java
package moe.sota.decompiler.types;

import moe.sota.decompiler.models.FileModel;
import org.fife.ui.rsyntaxtextarea.SyntaxConstants;
import org.jetbrains.annotations.NotNull;

public class ManifestType extends Type {

    public ManifestType() {
        super("icons/manifest.png", SyntaxConstants.SYNTAX_STYLE_PROPERTIES_FILE);
    }

    @Override
    public boolean isFormat(@NotNull FileModel fileModel) {
        return fileModel.getName().toLowerCase().endsWith(".mf");
    }

}

File: src/main/kotlin/moe/sota/decompiler/types/ManifestType.kt
package moe.sota.decompiler.types

import moe.sota.decompiler.models.FileModel
import org.fife.ui.rsyntaxtextarea.SyntaxConstants

class ManifestType :
    Type("icons/manifest.png", SyntaxConstants.SYNTAX_STYLE_PROPERTIES_FILE, true) {
    override fun isFormat(fileModel: FileModel): Boolean =
        fileModel.name.lowercase().endsWith(".mf")
}
